Add BMO scraper for big banks issue #7

hamzaelmi068 opened the pull request:
## 🏦 Add BMO Scraper for Big Banks Issue #7

###  What's Added:
- **BMO Scraper**: Complete scraper following RBC pattern
- **Scheduler Integration**: BMO scraper runs automatically  
- **API Endpoint**: `/api/scrape/bmo` for manual testing
- **Database Support**: PostgreSQL integration ready
- **Frontend Ready**: Company names will display correctly

## Testing:
-  scraper runs without errors
- api endpoint functional
- db should be working

### Next Steps:
- Research BMO website structure for better job detection
- Continue with other Big Banks (CIBC, TD, Scotiabank, etc.)
- Test with actual BMO job postings

faizm10 commented on backend/job-scraper/app/main.py:
```diff
@@ -183,7 +185,7 @@ async def get_stats(db: Session = Depends(get_db)):
 @app.post("/api/scrape")
 async def trigger_scrape(
-    company: str = Query("all", description="Company to scrape (all, microsoft, pinterest, rbc)"),
+    company: str = Query("all", description="Company to scrape (all, microsoft, pinterest, rbc, bmo)"),
```
remove pinterest. it is no longer associated with our app

hamzaelmi068 commented:
@faizm10 i made the changes you asked for:

- added the ontario/technology/intern filters to bmo scraper
- removed pinterest from the main endpoint 
- switched bmo to use playwright since their site is all javascript
- also added cibc scraper while i was at it

bmo scraper should work better now with the dynamic content. still testing to see if it finds those 3 jobs you mentioned.

got 4/6 banks done so far (bmo, cibc, rbc, microsoft). ready for you to check it out!

faizm10 commented:
LGTM!
